File: vouchergenerator/src/main/java/com/vouchergenerator/services/RecipientService.java
package com.vouchergenerator.services;

import com.vouchergenerator.entities.Recipient;
import com.vouchergenerator.form.RecipientForm;

public interface RecipientService {
    Recipient getRecipientByEmail(String email);
    RecipientForm newRecipient(RecipientForm form);
}

File: vouchergenerator/angularclient/src/app/services/recipient.service.ts
import { Injectable } from '@angular/core';
import {Router} from "@angular/router";
import {ApiRequestService} from "./api-request.service";

@Injectable({
  providedIn: 'root'
})
export class RecipientService {

  constructor(private router: Router,
              private apiService: ApiRequestService) { }

  login(body: Object) {
    return this.apiService.post("api/recipient/getRecipientByEmail", body);
  }

  getRecipientByID(body: Object) {
    return this.apiService.post("api/recipient/getRecipientByID", body);
  }

  register(body: Object) {
    return this.apiService.post("api/recipient/newRecipient", body);
  }

  getRecipientByEmail(body: Object) {
    return this.apiService.post("api/recipient/getRecipientByEmail", body);
  }

}
